src/main/scala/com/typesafe/sbt/SbtScalariform.scala

```diff
@@ -36,27 +42,43 @@ object SbtScalariform extends Plugin {
         prefixed("preferences"),
         "Scalariform formatting preferences, e.g. indentation"
       )
-
     private def prefixed(key: String) = s"scalariform-$key"
+
+    def scalariformSettings: Seq[Setting[_]] =
+      defaultScalariformSettings ++ List(
+        compileInputs in (Compile, compile) <<= (compileInputs in (Compile, compile)) dependsOn (format in Compile),
+        compileInputs in (Test, compile) <<= (compileInputs in (Test, compile)) dependsOn (format in Test)
+      )
   }
 
-  import ScalariformKeys._
+  import autoImport._
 
-  val defaultPreferences = {
-    import scalariform.formatter.preferences._
-    FormattingPreferences()
-      .setPreference(DoubleIndentClassDeclaration, true)
+  override lazy val projectSettings = scalariformSettings
+
+  override val trigger = allRequirements
+
+  object ScalariformKeys {
+
+    val format = autoImport.format
+
+    val preferences = autoImport.preferences
   }
 
-  def scalariformSettings: Seq[Setting[_]] =
-    defaultScalariformSettings ++ List(
+  def scalariformSettings: Seq[Setting[_]] = autoImport.scalariformSettings
+
+  def scalariformSettingsWithIt: Seq[Setting[_]] =
+    defaultScalariformSettingsWithIt ++ List(
       compileInputs in (Compile, compile) <<= (compileInputs in (Compile, compile)) dependsOn (format in Compile),
-      compileInputs in (Test, compile) <<= (compileInputs in (Test, compile)) dependsOn (format in Test)
+      compileInputs in (Test, compile) <<= (compileInputs in (Test, compile)) dependsOn (format in Test),
+      compileInputs in (It, compile) <<= (compileInputs in (It, compile)) dependsOn (format in It)
     )
 
   def defaultScalariformSettings: Seq[Setting[_]] =
     noConfigScalariformSettings ++ inConfig(Compile)(configScalariformSettings) ++ inConfig(Test)(configScalariformSettings)
 
+  def defaultScalariformSettingsWithIt: Seq[Setting[_]] =
+    defaultScalariformSettings ++ inConfig(It)(configScalariformSettings)
+
```

Comment:
Same as defaultScalariformSettings, you probably want to move this into autoImport.